graphql_compiler/interpreter/typedefs.py

```diff
@@ -172,8 +172,73 @@ def project_property(
+        the interpreter needs to get the value of some property field for a series of DataTokens.
+        For example, consider the following query:
+            {
+                Foo {
+                    bar @output(out_name: "bar_value")
+                }
+            }
+        Once the interpreter has used the get_tokens_of_type() function to obtain
+        an iterable of DataTokens for the Foo type, it will need to get the value of
+        the "bar" property on those tokens. This is where project_property() comes in.
+        The interpreter would call project_property() with arguments current_type_name = "Foo"
+        and field_name = "bar" along with an iterable of DataContext values, each of which
+        contains an optional DataToken as its current_token attribute. These DataContext
+        objects simply allow the interpreter to keep track of "which data came from where";
+        only the DataToken value within the current_token attribute is relevant to this function.
+        For each such DataToken held within a DataContext, project_property() needs to produce
+        the value of its property whose name is given as the field_name function argument.
+
+        A simple example implementation is as follows:
+            def project_property(
+                self,
+                data_contexts: Iterable[DataContext[DataToken]],
+                current_type_name: str,
+                field_name: str,
+                **hints: Any,
+            ) -> Iterable[Tuple[DataContext[DataToken], Any]]:
+                for data_context in data_contexts:
+                    current_token = data_context.current_token
+                    property_value: Any
+                    if current_token is None:
+                        # Evaluating an @optional scope where the optional edge didn't exist.
+                        # There is no value for the named property here.
+                        property_value = None
+                    else:
+                        property_value = < load the field_name property's value for current_token >
+
+                    # Remember to always yield the DataContext alongside the produced value
+                    yield data_context, property_value
+
+        Args:
+            data_contexts: iterable of DataContext objects which specify the DataTokens whose
+                           property data needs to be loaded
+            current_type_name: name of the vertex type whose property is being loaded. Guaranteed
+                               to be the name of a type defined in the schema being queried.
+            field_name: name of the property whose data to load. Guaranteed to refer to a property
+                        that is valid for the supplied current_type_name based on the schema.
+            runtime_arg_hints: names and values of any runtime arguments provided to the query
+                               for use in filtering operations (e.g. "$foo").
```

Comment:
Commented this on the `get_tokens_of_type` docstring PR as well, but maybe `baz` for the stand-in runtime arg since `Foo` and `bar` are already used in this docstring.